src/Psalm/Internal/Analyzer/Statements/Expression/ClassConstAnalyzer.php

```diff
@@ -206,7 +206,15 @@ public static function analyzeFetch(
             }
 
             if (!$stmt->name instanceof PhpParser\Node\Identifier) {
-                return true;
+                $was_inside_general_use = $context->inside_general_use;
+
+                $context->inside_general_use = true;
+
+                $ret = ExpressionAnalyzer::analyze($statements_analyzer, $stmt->name, $context);
```

Comment:
I suppose it should be version guarded, to make sure we don't allow this syntax when target version is below 8.3

Reply:
Also it's possible to access enum cases in the same way in 8.3, but it doesn't seem to be handled.

Reply:
Ah, I thought the version check would be handled by the parser, but apparently not. What behavior would you expect on a lower version than 8.3? `ParseError`?

Reply:
`ParseError` sounds good, as that's what PHP emits: https://3v4l.org/DfCB2